### PR TITLE
Playlist mgmt

### PR DESCRIPTION
**Pull request by Meghana-sk**
In this PR:
1. Playlist management
2. Private routes 
3.  Active link color added for Side navigation

-  Create new playlist if it already do not exist
- Delete playlist
- Add video to particular playlist
- Remove video from playlist

ps:
- Responsiveness not implemented

### CODE REVIEW

**Review by Ayushi20-19 — APPROVED** [review]
u can clear the input value after enter 
image

-else looks good 
-clean code u can merge, great work

File: src/components/side-nav/SideNav.jsx [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
-  backgroundColor: isActive ? "#2e3238" : null,
-  color: "whitesmoke",
-});
+const getActiveStyle = ({ isActive }) =>
+  isActive
+    ? {
+        color: "whitesmoke",
+        fontWeight: "bold",
+      }
+    : { color: "whitesmoke" };
```
Comment (gadgetfather) [inline src/components/side-nav/SideNav.jsx:10]: Avoid inline styling

File: src/pages/playlist/Playlist.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        }
+      );
+      if (response.status === 200) {
+        playlistDispatch({
+          type: UPDATE_PLAYLISTS,
+          payload: response.data.playlists,
+        });
+      }
+      toast.info(`${e.target.title} deleted successfully`);
+    } catch (error) {}
```
Comment (gadgetfather) [inline src/pages/playlist/Playlist.jsx]: You can also use toast to display error. 
`error.response` will give you object and then according to error code you can set custom message or  also response has predefined error message you can display that.

File: src/pages/playlist/PlaylistVideos.jsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        }
+      );
+      if (response.status === 200) {
+        playlistDispatch({
+          type: REMOVE_FROM_PLAYLISTS,
+          payload: response.data.playlist,
+        });
+        toast.info(`removed successfully`);
+      }
+    } catch (error) {}
```
Comment (gadgetfather) [inline src/pages/playlist/PlaylistVideos.jsx]: Same here ,if app fails to do certain action then user should know why it failed so use toast to display information.

File: src/shared/types.js [hunk 4]
```diff
@@ -22,3 +22,10 @@ const UPDATE_HISTORY = "UPDATE_HISTORY";
 const CLEAR_HISTORY = "CLEAR_HISTORY";
 
 export { UPDATE_HISTORY, CLEAR_HISTORY };
+
+/** Playlist variables */
+const UPDATE_PLAYLISTS = "UPDATE_PLAYLISTS";
+const ADD_TO_PLAYLISTS = "ADD_TO_PLAYLISTS";
+const REMOVE_FROM_PLAYLISTS = "REMOVE_FROM_PLAYLISTS";
+
+export { UPDATE_PLAYLISTS, ADD_TO_PLAYLISTS, REMOVE_FROM_PLAYLISTS };
```
Comment (gadgetfather) [inline src/shared/types.js:31]: Good work making them variables , saves a lots of time!

**Review by gadgetfather — APPROVED** [review]
- good file structure
- very good use of react fragment to avoid excessive div nesting 
- mostly try to avoid inline styling  , use classes
- don't forget to handle errors 

Apart from that great work! Playlist functionality is working well. You can add more videos in backend so while testing we can have more flexibility.

**Review by AnkurChunekar — APPROVED** [review]

**Pull request merged**